lib/src/Collection.cpp

```diff
@@ -247,11 +247,24 @@ pplx::task<shared_ptr<Document>> Collection::CreateDocumentAsync(
 	{
 		value json_response = response.extract_json().get();
 
-		if (response.status_code() == status_codes::Created)
-		{
-			return DocumentFromJson(json_response);
-		}
+		for (size_t i = 0; i < MAX_QUERY_RETRY_CNT; ++i) {

```

Comment:
I wanted to avoid handling 429 for a reason - I don't know what behavior user wanted, so only option was to just say to user "hey, here is 429, do what you want". Maybe they don't want to retry, maybe they want to quit their logic, or to lower number of threads (imagine user have N threads doing some work). That being said, I am OK with merging this code, even though 429 is not solved systematically (wouldn't it be better that this is wrappen in handle429() method that will know to retry?), but just for couple of calls. However, I cannot merge this in current state because it 1) changes existing behaviour, 2) doesn't give option to user to do what he/she wants. So, if you can please change this code, such that MAX_QUERY_RETRY_CNT is argument to these calls and is optional (=1), I will merge it. However, what would be proper solution is to offer user "strategy", which could be "do nothing/bail off", "retry N times", "exponential backoff N times with M seconds". But, this is OK for start, just please keep existing behaviour and move MAX_QUERY_RETRY_CNT from const to parameter to these functions

Reply:
사용자에게 '전략'을 선택할 기회를 줘야한다는 당신의 말이 맞다. 다만 당시에는 이 429 예외가 너무 쉽게 발생하였다. 최소 성능인 400RU/collection 뿐만 아니라 10000RU의 성능에서도 이 예외가 너무 많이 발생하였는데 테스트를 해보니 실제성능이 스펙성능보다 모자랐다. 어쨌거나 재시도 기능은 옵션으로 제공되도록 수정하겠다.

Reply:
Please. This is basically only reason why I am not merging this to master:)